app/src/test/resources/design-document.md

```diff
@@ -0,0 +1,118 @@
+U7. _As a user, I would like add a book to a new series I have started collecting_
+U8. _As a user, I would like create an account_
+U9 _As a user, I would like to login to my account_
+U10. _As a user, I would like to see collection based of my login_
+
+Stretch goals:
+- Search Option
+- Adding Images to the comics added(if wanted, user takes a picture and able to store that image)
+- Better design on website
+
+## Project Scope
+- view books that have been added to my collection
+- add books to a new series I am collection
+- view all books in my collection
+
+## Proposed Architecture Overview
+We will use API Gateway and Lambda to create endpoints()
+(GetAllComicBooksLambda, GetAllFavoriteLambda, GetAllStampsLambda, GetAllCoinsLambda, AddComicBookToCollectionLambda, AddStampsToCollectionLambda, AddCoinsToCollectionLambda, UpdateComicBookCollectionLambda, UpdateStampCollectionLambda, UpdateCoinCollectionLambda, SearchCollectionLambda)
+_that will handle the creation, update, and retrieval of events and vendors to satisfy our requirements_
+
+## API
+
+## 6.1 Public Models
+```
+// ComicBookModel
+
+String titleId
+String volumeNumber
+String issueNumber
+String date
+Int price
+Boolean isFavorite
+String publisher
+```
+
+## 6.2 _Get All Comic Books Endpoint_
+
+_* Accepts `GET` requests to `/comicbooks/`
+* Scans comics table and returns all ComicBookModels.
+    * If no ComicBookTable is found
+      `ComicBookNotFoundException`_
+
+
+## 6.3 _Get One ComicBook Endpoint_
+
+_* Accepts `GET` requests to `/comicbooks/:titleId/volumeNumber`
+* Accepts a comicbook ID and volumeNumber returns the corresponding Comic Book Title.
+    * If the given comicBook ID is not found or volumeNumber is not found, will throw a
+      `ComicBookNotFoundException`_
+
+## 6.4 _Add ComicBook Endpoint_
+
+* Accepts `POST` requests to `/comicbook/titleId/volumeNumber/add`
+* Accepts a comicbook ID and a ComicBookModel to be added.
+    * If the comicBook can not be added to the comicBookTable, will throw an `UnableToAddToTableException`
+
+
+## 6.5 _Update ComicBook Endpoint_
+
+* Accepts `PUT` requests to, `/comicbook/:titleId/volumenumber`
+    * If the given comicbook ID is not found and volumenumber, will throw a
+      `ComicBookNotFoundException`_
+
+## 6.6 _Get All Series Endpoint_
+
+_* Accepts `GET` requests to `/comicbooks/series`
+* Scans series table and returns all Series of pre-populated comic book model.
+    * If no ComicBookTable is found
+      `ComicBookNotFoundException`_
+
+## Tables
+Comic Books
+```
+seriesTitle // partition key, string Batman
+volumeNumber // sortkey, string 
+issueNumber // string
+date // string
+price // int, might use as a GSI, to find issues that I paid for a certain price. (might be a stretch goal)
+isFavorite // boolean, sort key for GSI to find if favorite
+publisher // string
+```
+
+
+Series - (pre-populated comic book table)
+
+```
+seriesTitle// partition key, string
+issues // string
```

Comment:
If this is going to be a list of issue names, the DynamoDb type Set (string set) would be a good option.